airflow/models/taskfail.py

```diff
@@ -39,6 +39,7 @@ class TaskFail(Base):
     duration = Column(Integer)
 
     __table_args__ = (
+        Index("idx_task_fail_task_instance", dag_id, task_id, run_id, map_index, unique=False),
```

Comment:
```suggestion
        Index("idx_task_fail_task_instance", dag_id, task_id, run_id, map_index),
```

That's the default, no?